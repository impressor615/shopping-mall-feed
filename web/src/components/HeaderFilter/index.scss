@import "~@/assets/styles/variables";

.header-filter {
  display: inline-block;

  .link:first-child {
    border-right: none;
    border-radius: 10px 0 0 10px;
  }

  .link:last-child {
    border-radius: 0 10px 10px 0;
  }
}