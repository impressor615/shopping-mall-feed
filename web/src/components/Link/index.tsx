import "./index.scss";

import classnames from "classnames";
import React from "react";
import { Link } from "react-router-dom";

interface ButtonProps {
  children: React.ReactNode;
  to: string;
  active?: boolean;
  [rest: string]: any;
}

const MyLink = ({ children, to, active, ...rest }: ButtonProps) => (
  <Link
    to={to}
    className={classnames("link", { active })}
    {...rest}
  >
    { children }
  </Link>
);

export default MyLink;
