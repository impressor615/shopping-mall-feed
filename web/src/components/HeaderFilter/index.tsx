import "./index.scss";

import React from "react";

import Link from "@/components/Link";

interface HeaderFilterProps {
  filter: string;
}

const HeaderFilter = ({ filter }: HeaderFilterProps) => (
  <div className="header-filter">
    <Link to="/products" active={filter === "products"}>상품</Link>
    <Link to="/ranks" active={filter === "ranks"}>랭킹</Link>
  </div>
);

export default HeaderFilter;
