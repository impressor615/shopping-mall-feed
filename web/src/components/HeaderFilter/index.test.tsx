import enzyme from "enzyme";
import React from "react";

import HeaderFilter from "@/components/HeaderFilter";
import Link from "@/components/Link";

describe("HeaderFilter", () => {
  const filter = "products";
  it("should render two buttons and one button should be active at least", () => {
    const wrapper = enzyme.shallow(<HeaderFilter filter={filter} />);
    expect(wrapper.find(Link)).toHaveLength(2);
    expect(wrapper.find(Link).first().props()).toHaveProperty("active", true);
  });

  it("should be matched with snapshot", () => {
    const wrapper = enzyme.shallow(<HeaderFilter filter={filter} />);
    expect(wrapper).toMatchSnapshot();
  });
});
