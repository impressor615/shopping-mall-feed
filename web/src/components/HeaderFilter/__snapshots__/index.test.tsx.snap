// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFilter should be matched with snapshot 1`] = `
<div
  className="header-filter"
>
  <MyLink
    active={true}
    to="/products"
  >
    상품
  </MyLink>
  <MyLink
    active={false}
    to="/ranks"
  >
    랭킹
  </MyLink>
</div>
`;
